a10_octavia/api/drivers/driver.py

```diff
@@ -304,3 +318,37 @@ def validate_flavor(self, flavor_dict):
                                   'due to: {}'.format(str(e)),
                 operator_fault_string='Failed to validate the flavor metadata '
                                       'due to: {}'.format(str(e)))
+
+    # Availability Zone
+    def get_supported_availability_zone_metadata(self):
+        """Returns the valid availability zone metadata keys and descriptions.
+
+        This extracts the valid availability zone metadata keys and
+        descriptions from the JSON validation schema and returns it as a
+        dictionary.
+
+        :return: Dictionary of availability zone metadata keys and descriptions
+        :raises DriverError: An unexpected error occurred.
+        """
+        props = {}
+        return props
+
+    def validate_availability_zone(self, availability_zone_dict):
```

Comment:
we could parent's class over here 
super().validate_availability_zone(availability_zone_dict)

Reply:
- I prefer the behavior control by our-own not up stream library
- To have our-own error message